code/products/mysql/connect.java

```diff
@@ -0,0 +1,40 @@
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Locale;
+
+public class MySqlExample {
+    public static void main(String[] args) throws ClassNotFoundException {
+        String host, port, databaseName, userName, password;
+        host = port = databaseName = userName = password = null;
+        for (int i = 0; i < args.length - 1; i++) {
+            switch (args[i].toLowerCase(Locale.ROOT)) {
+                case "-host": host = args[++i]; break;
+                case "-username": userName = args[++i]; break;
+                case "-password": password = args[++i]; break;
+                case "-database": databaseName = args[++i]; break;
+                case "-port": port = args[++i]; break;
+            }
+        }
+        // JDBC allows to have nullable username and password
+        if (host == null || port == null || databaseName == null) {
+            System.out.println("Host, port, database information is required");
+            return;
+        }
+        Class.forName("java.sql.Driver");
```

Comment:
in fact both ways are possible since `com.mysql.cj.jdbc.Driver` implements `java.sql.Driver` 
however in case of several drivers in same classpath probably makes sense

Reply:
Nope, `java.sql.Driver` is just an interface for JDBC drivers, loading it has no effect (since it is part of JDK anyway). Contrary, loading `com.mysql.cj.jdbc.Driver` adds `jdbc:mysq` JDBC MySQL connection handling, otherwise it won't work.